src/core/__tests__/template.test.tsx

```diff
@@ -652,4 +653,87 @@ describe("component/render in nested options", () => {
         component.update();
         expect(component.find(".template").html()).toBe('<div class="template">Second Template</div>');
     });
+
+    it("adds nested components dynamically", () => {
+        const renderItem = () => <div>Template</div>;
+        const items = [{id: 1, render: renderItem}];
+
+        const TestContainer = (props: any) => (
+            <TestComponent>
+                {props.items.map((item) => <CollectionNestedComponent key={item.id} render={item.render} />)}
+            </TestComponent>
+        );
+
+        const component = mount(<TestContainer items={items} />);
+        component.setProps({
+            items: [
+                ...items,
+                {id: 2, render: renderItem}
+            ]
+        });
+
+        const updatedOptions = Widget.option.mock.calls;
+
+        expect(updatedOptions[0][0]).toBe("integrationOptions");
+        expect(Object.keys(updatedOptions[0][1].templates)).toEqual([
+            "collection[0].template",
+            "collection[1].template"
+        ]);
+
+        expect(updatedOptions[1][0]).toBe("collection");
+        expect(updatedOptions[1][1].length).toBe(2);
+        expect(updatedOptions[1][1][0].template).toBe("collection[0].template");
+        expect(updatedOptions[1][1][1].template).toBe("collection[1].template");
+    });
+
+    it("removes nested components dynamically", () => {
+        const renderItem = () => <div>Template</div>;
+        const items = [{id: 1, render: renderItem}, {id: 2, render: renderItem}];
+
+        const TestContainer = (props: any) => (
+            <TestComponent>
+                {props.items.map((item) => <CollectionNestedComponent key={item.id} render={item.render} />)}
+            </TestComponent>
+        );
+
+        const component = mount(<TestContainer items={items} />);
+        component.setProps({
+            items: items.slice(0, 1)
+        });
+
+        const updatedOptions = Widget.option.mock.calls;
+
+        expect(updatedOptions[0][0]).toBe("integrationOptions");
+        expect(Object.keys(updatedOptions[0][1].templates)).toContain(
```

Comment:
Doesn't this assertion duplicate the one the next test has? (line [733](https://github.com/DevExpress/devextreme-react/pull/358/files#diff-266a9df62410a0a0b82ef942f935bea9R707)).
These two assertions are a bit different, but it seems that we should use only one of them, how do you think?

Reply:
Agree, fixed